napari/utils/action_manager.py

```diff
@@ -224,7 +225,13 @@ def bind_button(
                 )
```

Comment:
cannot include the lines in the suggestion so I will give the minor suggestion like this:
```
        if action := self._actions.get(name):
            if isgeneratorfunction(action):
                raise ValueError(
                    trans._(
                        '`bind_button` cannot be used with generator functions',
                        deferred=True,
                    )
                )
            else:
                button.setToolTip(
                    f'{self._build_tooltip(name)} {extra_tooltip_text}'
                )

        def _trigger():
            res = self.trigger(name)
            if isinstance(res, Generator):
                for _ in res:
                    pass

        button.clicked.connect(_trigger)
```

Reply:
Why would you like to raise an exception in such a case?

Reply:
In this case, the `isgeneratorfunction` check was already there. It originates from the discussion here: [#4164](https://github.com/napari/napari/issues/4164).

Reply:
Yes. I have fixed it. But the tooltip could be only set if an action is defined.